File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.py
from collections import deque

def solution(n, vertex: list):
    answer = 0
    result = [0 for _ in range(n)]
    
    visited = [False for _ in range(n)]
    visited[0] = True
    q = deque()
    q.append((1, 0))

    graph = {}

    for e1, e2 in vertex:
        if e1 not in graph:
            graph[e1] = []
        if e2 not in graph:
            graph[e2] = []

        graph[e1].append(e2)
        graph[e2].append(e1)


    while q:
        cur_edge, count = q.popleft()

        for edge in graph[cur_edge]:
            if not visited[edge-1]:
                visited[edge-1] = True
                q.append((edge, count + 1))
            else:
                result[cur_edge-1] = count

    answer = result.count(max(result))

    return answer

File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.swift
import Foundation

func solution(_ n:Int, _ edge:[[Int]]) -> Int {
    
    var visited: [Int] = .init(repeating: -1, count: n + 1)
    var board: [[Int]] = .init(
        repeating: [],
        count: n + 1
    )
    
    for e in edge {
        board[e[0]].append(e[1])
        board[e[1]].append(e[0])
    }
    
    var queue: [Int] = [1]
    visited[1] = 0
    var front = 0
    var maximum = 0
    
    while queue.count >= front + 1 {
        let curNode = queue[front]
        front += 1
        let curDist = visited[curNode]
        maximum = max(maximum, curDist)
        
        for nextNode in board[curNode] {
            if visited[nextNode] == -1 { 
                queue.append(nextNode)
                visited[nextNode] = curDist + 1
            }    
        }
    }
    
    return visited.filter { $0 == maximum }.count
    
}